src/ui/services/ipc/ipcRendererConnection.ts

```diff
@@ -1,25 +1,32 @@
-import { IpcChannel, IpcChannelMessage } from '../../../common/ipc';
+import {
+  IpcChannel,
+  IpcChannelMessage,
+  IpcChannelResponse,
+} from '../../../common/ipcMain';
 
 /*
  * Responsible for all communications with the main process.
  */
 export class IpcRendererConnection {
-  constructor() {
-    window.electron.ipcRenderer.on(
-      IpcChannel.MAIN_UTILS_PING_REPLY,
-      this.onMainPingReply
-    );
-  }
-
   private sendMessage<T extends IpcChannel>(
     channel: T,
     message: IpcChannelMessage<T>
   ): void {
     window.electron.ipcRenderer.sendMessage(channel, message);
   }
 
-  mainPing(): void {
-    this.sendMessage(IpcChannel.MAIN_UTILS_PING, { data: { message: 'ping' } });
+  private async invoke<T extends IpcChannel>(
+    channel: T,
+    message: IpcChannelMessage<T>
+  ): Promise<IpcChannelResponse<T>> {
+    return window.electron.ipcRenderer.invoke(channel, message);
+  }
+
+  async mainPing(): Promise<void> {
+    const message = await this.invoke(IpcChannel.MAIN_UTILS_PING, {
```

Comment:
All of this PR was made for this line. We can await for a response instead of registering a callback separately.